.github/workflows/release.yml

```diff
@@ -79,3 +79,31 @@ jobs:
           files: |
             core/target/google-java-format*
             eclipse_plugin/target/google-java-format-eclipse-plugin-*.jar
+
+  build-native-image:
+    name: "Build GraalVM native-image on ${{ matrix.os }}"
+    runs-on: ${{ matrix.os }}
+    needs: build-maven-jars
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+    steps:
+      - name: "Check out repository"
+        uses: actions/checkout@v4
+      - name: "Set up GraalVM"
+        uses: graalvm/setup-graalvm@v1
+        with:
+          java-version: "21"
+          distribution: "graalvm-community"
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          native-image-job-reports: "true"
+          cache: "maven"
+      - name: "Native"
+        run: mvn -Pnative -DskipTests package -pl core -am
+      - name: "Move outputs"
+        run: cp core/target/google-java-format google-java-format-${{ matrix.os == 'ubuntu-latest' && 'linux' || 'darwin' }}
```

Comment:
@cushon I don't understand this expression... and it (now) won't work for Windows?

Reply:
It returns the string 'linux' or 'darwin' depending on the current OS, so the platform-specific outputs get uploaded with unique names. The syntax is documented here: https://docs.github.com/en/actions/learn-github-actions/expressions

I agree we'll want to update this for windows, I'm not sure what the cleanest way to do that is yet. I looked around at a few different ways to upload these artifacts to the release and didn't find anything very appealing.

Reply:
Oh, I think I just got it - the idea is that this will evaluate to the following, right?

* `google-java-format-linux`
* `google-java-format-darwin`

How about we use slightly different suffixes for the released native binaries? Inspired by e.g. `uname -a` output, I would suggest:

* `google-java-format_linux-x86_64`
* `google-java-format_darwin-arm64` (or something like that - what exactly does `uname -a` output on your Mac?)
* `google-java-format_win-x64` (I think that's what that's commonly referred to - but I'm not 100% sure)

Reply:
Those suffixes LGTM, thanks. (My uname output includes `Darwin` and `arm64`.)

Reply:
> I agree we'll want to update this for windows, I'm not sure what the cleanest way to do that is yet. I looked around at a few different ways to upload these artifacts to the release and didn't find anything very appealing.

Simplest could be to duplicate this (and the "gh release upload") lines x3, with an "if" for penguins, apples, and wins?

Reply:
> * `google-java-format_win-x64` (I think that's what that's commonly referred to - but I'm not 100% sure; but I would _NOT_ use "x86_64" à la Linux for this binary, because MS seems to use x86=32bit VS x64=64bit - apparently.)

Just noticed on https://github.com/bazelbuild/bazel/releases that they use "*windows_x86-64" so perhaps we can go for that here as well after all.

Reply:
Thanks! I found a different way to express that mapping, and updated to use the suggested names.